primitives/core/src/lib.rs

```diff
@@ -78,6 +77,42 @@ impl From<MessageSendError> for &'static str {
 	}
 }
 
+/// The origin of an inbound message.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
+pub enum AggregateMessageOrigin {
+	/// The message came from the para-chain itself.
+	Loopback,
+	/// The message came from the relay-chain.
+	///
+	/// This is used by the DMP queue.
+	Parent,
+	/// The message came from a sibling para-chain.
+	///
+	/// This is used by the HRMP queue.
+	Sibling(ParaId),
+}
+
+impl From<AggregateMessageOrigin> for xcm::v3::MultiLocation {
+	fn from(origin: AggregateMessageOrigin) -> Self {
+		match origin {
+			AggregateMessageOrigin::Loopback => MultiLocation::here(),
+			AggregateMessageOrigin::Parent => MultiLocation::parent(),
+			AggregateMessageOrigin::Sibling(id) => Junction::Parachain(id.into()).into(),
```

Comment:
```suggestion
			AggregateMessageOrigin::Sibling(id) => MultiLocation::new(1, Junction::Parachain(id.into())),
```

Should probably changed as suggested?

Reply:
Yep looks better. Thanks!